Client/Frontend/Widgets/TwoLineCell.swift

```diff
@@ -187,21 +220,38 @@ private class TwoLineCellHelper {
         detailTextLabel.font = UIFont.systemFontOfSize(DynamicFontHelper.defaultHelper.DefaultSmallFontSize, weight: UIFontWeightRegular)
     }
 
+    func setupHistoryPanelDynamicFonts() {
+        textLabel.font = DynamicFontHelper.defaultHelper.DeviceFontHistoryPanel
+        detailTextLabel.font = DynamicFontHelper.defaultHelper.DeviceFontSmallHistoryPanel
+    }
+
     func layoutSubviews() {
         let height = container.frame.height
-        let textLeft = ImageSize + 2 * ImageMargin
+        let textLeft = borderView != nil ? TwoLineCellUX.BorderWidth + 2 * TwoLineCellUX.BorderViewMargin : TwoLineCellUX.ImageSize + 2 * TwoLineCellUX.BorderViewMargin
         let textLabelHeight = textLabel.intrinsicContentSize().height
         let detailTextLabelHeight = detailTextLabel.intrinsicContentSize().height
         var contentHeight = textLabelHeight
         if detailTextLabelHeight > 0 {
-            contentHeight += detailTextLabelHeight + DetailTextTopMargin
+            contentHeight += detailTextLabelHeight + TwoLineCellUX.DetailTextTopMargin
         }
-        
-        imageView.frame = CGRectMake(ImageMargin, (height - ImageSize) / 2, ImageSize, ImageSize)
+
+        if let borderView = borderView {
+            borderView.frame = CGRectMake(TwoLineCellUX.BorderViewMargin, TwoLineCellUX.BorderViewMargin, TwoLineCellUX.BorderWidth, TwoLineCellUX.BorderWidth)
+            borderView.layer.borderWidth = 0.5
+            borderView.layer.borderColor = UIColor.grayColor().CGColor
+
+            imageView.snp_makeConstraints { make in
+                make.center.equalTo(borderView)
+                make.size.equalTo(TwoLineCellUX.ImageSize)
+            }
+        } else {
+            imageView.frame = CGRectMake(TwoLineCellUX.BorderViewMargin, (height - TwoLineCellUX.ImageSize) / 2, TwoLineCellUX.ImageSize, TwoLineCellUX.ImageSize)
+        }
+
         textLabel.frame = CGRectMake(textLeft, (height - contentHeight) / 2,
-            container.frame.width - textLeft - ImageMargin, textLabelHeight)
-        detailTextLabel.frame = CGRectMake(textLeft, textLabel.frame.maxY + DetailTextTopMargin,
-            container.frame.width - textLeft - ImageMargin, detailTextLabelHeight)
+            container.frame.width - textLeft - TwoLineCellUX.BorderViewMargin, textLabelHeight)
+        detailTextLabel.frame = CGRectMake(textLeft, textLabel.frame.maxY + TwoLineCellUX.DetailTextTopMargin,
+            container.frame.width - textLeft - TwoLineCellUX.BorderViewMargin, detailTextLabelHeight)
```

Comment:
Yikes. This layout is turning into a hot mess.

If possible, I'd avoid leaking the `borderView` into `TwoLineCellHelper`. You only need a border in your `HistoryTableViewCell`, so create it and do the layout there. I'd suggest doing the normal layout pass, then adjusting the border around the image as part of `HistoryTableViewCell`'s `layoutSubviews`. [Here's a quick example.](https://gist.github.com/thebnich/8168fc2666c961043042f9595b864290)